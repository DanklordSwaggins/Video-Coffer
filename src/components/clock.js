import React from "react";
import moment from "moment";
import "../styles/clock.css";

class Clock extends React.Component {
  constructor(props) {
    super(props);
    this.state = { time: new moment().format("LLL").toLocaleString() };
  }

  tick() {
    this.setState({ time: new moment().format("LLL").toLocaleString() });
  }

  componentDidMount() {
    this.intervalID = setInterval(() => this.tick(), 1000);
  }

  componentWillUnmount() {
    clearInterval(this.intervalID);
  }

  render() {
    return <h6 className="clock">{this.state.time}</h6>;
  }
}

export default Clock;
