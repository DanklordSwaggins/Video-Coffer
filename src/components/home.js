import React from 'react';
import '../styles/home.css';
import SearchBar from './searchBar';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
import {history} from '../routes';

const toplayer = ()=>{
        //ReactDOM.render(jsx, document.getElementById('root'));
        history.push('/videoPlayer')
};

export const Home = ({startLogout}) => {
    return (
        <div className="Home">
            <div className="lander">
                <h1>Video Coffer</h1>
                <p>If you love a video, don't let it go</p>
                <SearchBar />
                <div>
                    <button onClick={function(event){ startLogout()}}>Logout</button>
                    <button onClick={function(event){ toplayer()}}>player</button>
                </div>
            </div>
        </div>
    );
};

const mapDispatchToProps=(dispatch)=>({
    startLogout: () =>dispatch(startLogout())
});

export default connect(undefined,mapDispatchToProps)(Home);