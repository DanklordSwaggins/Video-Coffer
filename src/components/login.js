import React, { useState } from 'react';
import { Button, Form } from 'react-bootstrap';
import '../styles/login.css';

const Login = (props) => {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");

    const validateForm = () => {
        return email.length > 0 && password.length > 0;
    }

    const handleSubmit = (event) => {
        event.preventDefault();
    }

    return (
        <div className="Login">
            <Form onSubmit={handleSubmit}>
                <Form.Group controlId="formGroupEmail">
                    <Form.Label>Email Address</Form.Label>
                    <Form.Control
                    autoFocus
                    type="email"
                    value={email}
                    placeholder="Enter Email"
                    onChange={e => setEmail(e.target.value)}
                    />
                </Form.Group>
                <Form.Group controlId="formGroupPassword">
                    <Form.Label>Password</Form.Label>
                    <Form.Control
                    type="password"
                    value={password}
                    placeholder="Password"
                    onChange={e => setPassword(e.target.value)}
                    />
                </Form.Group>
                <Button
                block
                variant="primary"
                disabled={!validateForm()}
                type="submit"
                >
                Login
                </Button>
            </Form>
        </div>
    );*/

    
    

    return(
      <div>
        <Button
        onClick={fireAuth} 
        >
          Sign in with google
        </Button>
      </div>
    );
      
}

    

export default Login;