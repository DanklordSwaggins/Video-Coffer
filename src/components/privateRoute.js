import React, {useEffect, useState} from 'react';
import {firebase} from '../firebase/firebaseConfig';


// const privateRoute = props => {
//     const [authenticated, setAuthenticatedStatus] = 
//     useEffect(() => {
//         firebase.auth().onAuthStateChanged((user)=>{ 
//             if(user){
//               //console.log('logged in')
//               //window.location.replace('/home');
//               // set state
//               history.push('/home');
//             }else{
//               history.push('/');
//             }
//           });
//     }, [])

//     return (

//     )
// }
