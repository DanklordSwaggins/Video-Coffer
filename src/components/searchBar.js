import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { InputGroup, FormControl, Spinner, Card, Button, CardDeck } from 'react-bootstrap';
import { debounce } from 'throttle-debounce';
import videoSearch from '../actions/videos';
import videoId from '../actions/videoId';
import '../styles/searchBar.css';


const SearchBar = () => {

    const dispatch = useDispatch();
    const videos = useSelector(state => state.videos);
    const { fetched, fetching, error } = videos;

    let history = useHistory();

    const handleOnChange = debounce(700, (e) => {
        dispatch(videoSearch({ keywords: e, maxResults: 50, pageToken: '' }))
    });

    const handleClick = (key) => {
        dispatch(videoId(key))
        history.push('/videoPlayer')
    };

    return (
        <div>
            <InputGroup size="lg">
                <FormControl
                    placeholder="Inject queries here..."
                    onChange={(value) => handleOnChange(value.target.value)}
                />
            </InputGroup>
            <div className="flex-container">
                {fetching &&
                    <Spinner
                        animation="border"
                        variant="danger"
                        role="status"
                    />
                }
                {error ? <p>We encountered an error :(</p> : null}
                {fetched && videos.items.map((item) => {
                    return (
                        <div key={item.id.videoId}>
                            <CardDeck className="results">
                                <Card border="danger" className="cardItem">
                                    <Card.Img variant="top" src={item.snippet.thumbnails.medium.url} />
                                    <Card.Body>
                                        <Card.Title>{item.snippet.title}</Card.Title>
                                        <Card.Text>{item.snippet.description}</Card.Text>
                                    </Card.Body>
                                    <Button
                                        variant="danger"
                                        onClick={() => { handleClick(item.id.videoId) }}
                                    >Play</Button>
                                    <Button variant="warning">Watch Later</Button>
                                </Card>
                            </CardDeck>
                        </div>
                    )
                })}
            </div>
        </div>
    );
};

export default SearchBar;