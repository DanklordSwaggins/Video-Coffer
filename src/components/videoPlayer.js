import React from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { Button } from 'react-bootstrap';
import '../styles/videoPlayer.css';

const VideoPlayer = () => {

    const videoId = useSelector(state => state.videoId);
    const { id } = videoId;

    let history = useHistory();

    const handleClick = () => {
        history.push('/home');
    }

    return (
        <div className="container">
            <div className="mediaPlayer">
                <Button
                    variant="primary"
                    onClick={handleClick}
                >Go Back</Button>
                <iframe
                    src={`https://www.youtube.com/embed/${id}`}
                    frameBorder='0'
                    height='480'
                    width='640'
                    allow='autoplay; encrypted-media'
                    allowFullScreen
                    title='video'
                />
            </div>
        </div>
    );
};

export default VideoPlayer;