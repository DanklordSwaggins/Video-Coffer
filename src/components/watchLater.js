import React from "react";
import "../styles/watchLater.css";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../actions/auth";
import { removeVideoList } from "../actions/videoId";
import { Card, ButtonGroup, Button, CardDeck } from "react-bootstrap";
import videoId from "../actions/videoId";

export const WatchLater = () => {

  const history = useHistory();
  const dispatch = useDispatch();

  const videolist = useSelector(state => state.list);
  const sName = useSelector(state => state.users);
  const { email } = sName;

  const toPlayer = key => {
    dispatch(videoId(key));
    history.push("/videoPlayer");
  };

  const HandlePush = item => {
    const videoData = {
      sid: item.sid,
      email: email
    };
    dispatch(removeVideoList(videoData));
    history.push("/watchLater");
  };

  return (
    <div className="WatchLater">
      <div className="d-flex flex-column">
        <ButtonGroup>
          <Button variant="warning" onClick={() => history.push("/videoPlayer")}>Video Player</Button>
          <Button variant="warning" onClick={() => history.push("/home")}>Home</Button>
          <Button variant="warning" onClick={() => { dispatch(startLogout()) }}>Logout</Button>
        </ButtonGroup>
      </div>
      <div className="lander">
        <h1>Video Coffer</h1>
        <p>If you love a video, don't let it go</p>
        <div className="flex-container">
          {videolist.map(item => {
            return (
              <div key={item.id}>
                <CardDeck className="results">
                  <Card border="danger" className="cardItem">
                    <Card.Img variant="top" src={item.thumbnail} />
                    <Card.Body>
                      <Card.Title>{item.title}</Card.Title>
                      <Card.Text>{item.description}</Card.Text>
                    </Card.Body>
                    <Button
                      variant="danger"
                      onClick={() => { toPlayer(item.id) }}>Play</Button>
                    <Button
                      variant="warning"
                      onClick={() => { HandlePush(item) }}>Remove from Watch List</Button>
                  </Card>
                </CardDeck>
              </div>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default WatchLater;