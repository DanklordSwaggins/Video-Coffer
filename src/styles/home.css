.Home .lander {
  padding: 80px 20px;
  text-align: center;
}

.Home .lander h1 {
  font-family: "Open Sans", sans-serif;
  font-weight: 600;
}

.Home .lander p {
  color: #999;
}

.Home .results {
  margin-top: 2rem;
}

.Home .buttons {
  margin: 1rem 0rem;
}
