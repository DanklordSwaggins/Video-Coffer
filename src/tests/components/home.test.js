import React from 'react';
import {shallow, mount} from 'enzyme';
import {Home} from '../../components/home';
import configureStore from '../..//store/store'
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";


it('should render home', ()=>{
    const store = configureStore();
     const wrapper = mount(
        <Provider store={store}>
        <Router>
        <Home/>
        </Router></Provider>
    );
    expect(wrapper).toMatchSnapshot();

})

