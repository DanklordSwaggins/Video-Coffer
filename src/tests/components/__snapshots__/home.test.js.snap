// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render home 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Home>
        <div
          className="Home"
        >
          <div
            className="d-flex flex-column"
          >
            <ButtonGroup
              role="group"
              toggle={false}
              vertical={false}
            >
              <div
                className="btn-group"
                role="group"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="warning"
                >
                  <button
                    className="btn btn-warning"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Video Player
                  </button>
                </Button>
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="warning"
                >
                  <button
                    className="btn btn-warning"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    My List
                  </button>
                </Button>
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="warning"
                >
                  <button
                    className="btn btn-warning"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Logout
                  </button>
                </Button>
              </div>
            </ButtonGroup>
          </div>
          <div
            className="lander"
          >
            <h1>
              Video Coffer
            </h1>
            <p>
              If you love a video, don't let it go
            </p>
            <SearchBar>
              <div>
                <InputGroup
                  size="lg"
                >
                  <div
                    className="input-group input-group-lg"
                  >
                    <FormControl
                      onChange={[Function]}
                      placeholder="Inject queries here..."
                    >
                      <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="Inject queries here..."
                      />
                    </FormControl>
                  </div>
                </InputGroup>
                <div
                  className="flex-container"
                />
              </div>
            </SearchBar>
          </div>
        </div>
      </Home>
    </Router>
  </BrowserRouter>
</Provider>
`;
